script/model/training_util/data_transforming_util.py

```diff
@@ -90,18 +91,53 @@ def _tuple_num_linear_log_predict_transform(x, y):
 def _tuple_num_log_train_transform(x, y):
     # Transform down the target according to the log tuple num value in the input
     tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
-    return y / (np.log2(tuple_num) + 1)[:, np.newaxis]
+    return y / (np.log2(tuple_num) + _TRANSFORM_EPSILON)[:, np.newaxis]
 
 
 def _tuple_num_log_predict_transform(x, y):
     # Transform up the target according to the log tuple num value in the input
     tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
-    return y * (np.log2(tuple_num) + 1)[:, np.newaxis]
+    return y * (np.log2(tuple_num) + _TRANSFORM_EPSILON)[:, np.newaxis]
 
 
 # Transform the target in a log scale (logn) according to the tuple num
 _tuple_num_log_transformer = (_tuple_num_log_train_transform, _tuple_num_log_predict_transform)
 
+
+def _tuple_num_cardinality_linear_log_train_transform(x, y):
+    # Transform down the target according to the cardinality * log tuple num value in the input
+    tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
+    cardinality = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.EST_CARDINALITIES]])
+    return y / ((cardinality * np.log2(tuple_num)) + _TRANSFORM_EPSILON)[:, np.newaxis]
+
+
+def _tuple_num_cardinality_linear_log_predict_transform(x, y):
+    # Transform up the target according to the cardinality * log tuple num value in the input
+    tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
+    cardinality = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.EST_CARDINALITIES]])
+    return y * ((cardinality * np.log2(tuple_num)) + _TRANSFORM_EPSILON)[:, np.newaxis]
+
+
+# Transform the target in a linear log scale (cardinality * log tuple_num)
+_tuple_num_cardinality_linear_log_transformer = (_tuple_num_cardinality_linear_log_train_transform,
+                                                 _tuple_num_cardinality_linear_log_predict_transform)
+
+def _tuple_cardinality_linear_train_transform(x, y):
```

Comment:
Remove the "tuple" and just call this  `_cardinality_linear_train_transform`. Apply to others as well.